src/cargo/util/to_url.rs

```diff
@@ -40,3 +60,38 @@ fn mapper(s: &str) -> url::SchemeType {
         s => url::whatwg_scheme_type_mapper(s),
     }
 }
+
+fn determine_base_url() -> Result<Url, String> {
+    let current_dir = try!(env::current_dir().map_err(|s| {
```

Comment:
It seems like it should be relative to the file that contains the URL rather than the directory cargo is invoked in?